packages/supplier/src/supplier.ts

```diff
@@ -0,0 +1,181 @@
+import type { Strategy as FreeStrategy, BaseStrategy } from "./strategy";
+import type { Results as FreeResults } from "./results";
+
+/**
+ * Given the definition of specified supplier, create a function that
+ * determines+constructs a supplier strategy for a given set of options.
+ */
+export const forDefinition = <S extends Supplier.Specification>({
+  determineStrategy,
+  strategyConstructors
+}: Supplier.Definition<S>): CreateSupplier<S> => <
+  N extends Supplier.StrategyName<S>
+>(
+  options: Supplier.Strategy.Constructor.Options<S, N>
+): Supplier<S, N> => {
+  const strategyName = determineStrategy(options);
+  const Strategy = strategyConstructors[strategyName];
+
+  // @ts-ignore since we can't figure out N from inside
```

Comment:
I'm of the mind now that `@ts-ignore` is safer than coercions

Reply:
I strongly disagree with this point of view.  I do not think we should be using `ts-ignore`... basically at all.  We used it in that one place in `encoder` only because the configuration inconsistency (where the same code had to compile with `strictNullChecks` both on and off) left us with basically no other choice.

In general, we should try to work within TS, not ignore it.  Obviously coercions are not fully "working within TS", but they're closer.  TS does for instance perform rudimentary checks on coercions to make you're not sure you're doing anything too stupid with them, for instance; it obviously doesn't do that with ts-ignore.

You have elsewhere, although not here, made arguments as to why coercion is unsafe.  I do not see that any of these arguments apply here.  Indeed, I'm not convinced those arguments apply much of anywhere, as you've never provided any concrete examples, and the ones you've vaguely described seem to me to rely on what I would consider to be other bad practices; I would say those are the problems, not the coercion.

If you want to convince me on this point you are going to need to provide much more in the way of argument.

I would strongly discourage the use of ts-ignore both here and everywhere.